app/models/access_control.rb

```diff
@@ -18,11 +18,15 @@ def can?(user, action, resource_namespace, scope = nil)
         case action
         when :read then true
         when :write
-          if scope
-            user.deployer_for?(scope) # stage locks
+
+          if scope.nil? # global locks
+            user.admin?
+          elsif ENV['PRODUCTION_STAGE_LOCK_REQUIRES_ADMIN'] && scope.production
```

Comment:
prefer to guard this in a `elsif scope.is_a?(Stage)`so future additions don't require a refactor